cmd/mimir/main.go

```diff
@@ -138,6 +150,11 @@ func main() {
 		}
 	}
 
+	if mainFlags.verifyConfig {
```

Comment:
It's admirable to keep consistency with Loki but I'm confused why we need to add two flags. Could we move the `if mainConfig.printConfig {` check after validation and have it exit?